src/crawlee/basic_crawler/_basic_crawler.py

```diff
@@ -524,6 +526,14 @@ async def add_requests(
             wait_for_all_requests_to_be_added_timeout=wait_for_all_requests_to_be_added_timeout,
         )
 
+    async def _use_state(self, key: str, default_value: dict | None = None) -> dict:
```

Comment:
General dicts (`dict`) are used here, but more specific (`dict[str, Any]`) are used in the KVS. Is that intended?

Reply:
Yes, it's a state `dict`, it is the `Any` in KVS `dict`

Reply:
Thank you. Matched the types between KVS and the function

Reply:
What is the reasoning behind using a plain `dict`? Also, we might want to use `dict[str, JsonSerializable]` - otherwise it might confuse users (like `Request.user_data` did).

Reply:
I like the idea of using `dict[str, JsonSerializable]` it makes the signature and usage clearer.